build/cmake/setup.cmake

```diff
@@ -587,12 +587,20 @@ if(wxUSE_FSWATCHER)
 endif()
 
 if(wxUSE_XLOCALE)
+    list(APPEND xlocale_headers locale.h stdlib.h)
     check_include_file(xlocale.h HAVE_XLOCALE_H)
-    set(CMAKE_EXTRA_INCLUDE_FILES locale.h)
     if(HAVE_XLOCALE_H)
-        list(APPEND CMAKE_EXTRA_INCLUDE_FILES xlocale.h)
+        list(APPEND xlocale_headers xlocale.h)
     endif()
-    check_type_size(locale_t LOCALE_T)
+    wx_check_c_source_compiles("
+        locale_t t;
+        strtod_l(NULL, NULL, t);
+        strtol_l(NULL, NULL, 0, t);
+        strtoul_l(NULL, NULL, 0, t);
+"
+        HAVE_LOCALE_T
+        ${xlocale_headers}
+        )
     set(CMAKE_EXTRA_INCLUDE_FILES)
```

Comment:
You can get rid of this line too. `CMAKE_EXTRA_INCLUDE_FILES` is not used anymore so no need to reset it.

Reply:
Sorry, I planned to do it but somehow forgot. I'll do it when merging (if you don't have any further comments).

Reply:
No other comments. `CMake builds / Ubuntu 22.04` still passes the test, so looks good to me.